### ISSUE TITLE
Show only associated places for Online users on LHS in web app.

### ISSUE THREAD

**Issue by yrimal** [medic/cht-core#7496]
**Is your feature request related to a problem? Please describe.**
For online users who belong to particular place, all the places on the system are shown on LHS. 

**Describe the solution you'd like**
For a online User, who belongs to `Area A`, I want to show places under `Area A` only show that only `Area A-1, Area A-2` etc are shown for selection on LHS. I don't want to see places under any other area where current online logged in user is not contact person to.

**Comment by mrjones-plip** [comment]
@michaelkohn  - just added this to care teams - pinging you as a heads up!

**Comment by garethbowen** [comment]
@yrimal So that we can scope this correctly, is there a permissions component, ie: should the user not be allowed to access or modify places outside of their area? Or is this simply a user convenience so they can more quickly get to the contacts they are interested in?

**Comment by yrimal** [comment]
Hi @garethbowen ,
Priority for now is user convenience, because all subunits under CHOs are named in the same pattern (Ward 1, Ward 2, Ward 3,  etc). Hence, If we have 10 CHOs, they'll see 10 Ward 1, 10 Ward 2, 10 Ward 3s and they do not know which ward is theirs and fall in trap. 

![Screen Shot 2022-02-02 at 9 27 16 PM](https://user-images.githubusercontent.com/6102813/152187478-dfc5ce47-11af-4ed7-a7e4-e29f58fc16ea.png)

Requirement now is for CHOs to be able to go to their respective sub units without multiple guesses. 

>is there a permissions component? 
This will be nice to have configurable option, but not until now. But I see this request coming sooner or later from government officers.

**Comment by michaelkohn** [comment]
@yrimal Conceptually speaking, would it solve their problem if they were simply able to _see_ the parent? For example...

> Ward 1 (Municipality 1)
> Ward 1 (Municipality 2)
> Ward 1 (Municipality 3)
> Ward 2 (Municipality 1)
> Ward 2 (Municipality 2)

Also, do you have an indication of how many places they will have at each level? The reason I ask is that once they tap on the Ward, it will take them to the **_Contact_** page of that Ward and show a list of CHN Areas (in the selected Ward). From the contacts page of the selected CHN Area they'll see a list of FCHV Areas, etc, etc. The Contacts page doesn't take advantage of lazy loading and doesn't have search capabilities, so it's important to make sure the [user experience will be acceptable as they drill down in the hierarchy](https://github.com/medic/cht-core/issues/7459#issuecomment-1013443072).

**Comment by yrimal** [comment]
Hi @michaelkohn , for now, putting municpality name on bracket would make them pick correct ward. But, this would also be a problem if we scale up from two municipalties to 5/10 municpalties. They may struggle finding their Ward 1 (Municipality) because even with just 5 municipalities, there'd be 15x5 = 75 wards. 

>do you have an indication of how many places they will have at each level?

On an average under one CHO, there will be 10-20 Wards (i.e. 10-20 CHNs). Under each ward, there can be minimum 10 to up to 25/30 FCHVs. Under those FCHV area will be households and so on.

**Comment by michaelkohn** [comment]
Got it, thanks @yrimal , I'll follow up in Slack for some more project specific questions.

**Comment by michaelkohn** [comment]
There were some calls and conversations elsewhere.

- [Forum post re: what online users see](https://forum.communityhealthtoolkit.org/t/what-data-do-online-users-see/1694)
- [Meeting notes to discuss specific project details](https://docs.google.com/document/d/1ctpEE0y9q860ObfHJD2Xgu840JyXn8QcPHg0JvhAaUA/edit#)
- [Slack thread](https://medic.slack.com/archives/C4LVCKJJC/p1644330945348969)

Summary of latest questions @yrimal :

1. What kinds of things / errors will CHNs be looking for in the forms
2. Are there any calculations / formulas that can help identify those issues in # 1 above
3. Will CHOs be physically with CHNs during a supervisory meeting or will the CHNs be doing this totally independently
4. When CHNs make edits to forms, will they be notifying the CHOs of the edits / errors
5. Is there any indication today of how often this will need to happen
6. And just to confirm, above you mentioned that this is primarily for convenience rather than an issue with accessing data in other areas, is that still correct?

**Comment by yrimal** [comment]
**1. What kinds of things / errors will CHNs be looking for in the forms**
Basically all the information on the submitted form and review for any errors (incorrect option selected) and factfulness of the submitted form so that they can suggest continuous care guides based on the data collected. 

**2. Are there any calculations / formulas that can help identify those issues in # 1 above.**
There's no formula. But, CHOs will review, correlate an information collected and if in doubt, talk with CHNs (through phone immediately if needed). Also there can be additional guidance to be provided based on the information collected. Hence, reviewing and looking submitted reports is CHO's primary duty. 

**3. Will CHOs be physically with CHNs during a supervisory meeting or will the CHNs be doing this totally independently**

CHOs duty is to supervise the information collected, so whenever new forms are submitted, they review them. CHOs meet CHNs physically from daily basis to weekly basis.

**4. When CHNs make edits to forms, will they be notifying the CHOs of the edits / errors**

Yes.

**5. Is there any indication today of how often this will need to happen**

Every day, it's day to day work of CHOs in areas where an app has been deployed. It's not periodic need. 

**6. And just to confirm, above you mentioned that this is primarily for convenience rather than an issue with accessing data in other areas, is that still correct?**

No, it's both. CHOs should not be able to see data in any other areas. For now, we only have two municpalities that's why it doesn't sound as inconvenient and data issue. But, we've scale up in June/July and then it'll be a major problem. The requirement is to see data under them only.

**Comment by michaelkohn** [comment]
> **1. What kinds of things / errors will CHOs be looking for in the forms**
Basically all the information on the submitted form and review for any errors (incorrect option selected) and factfulness of the submitted form so that they can suggest continuous care guides based on the data collected.

Do you have specific examples of the types of things they might find? How will the CHO know if something is wrong on the form just by looking at it (and not talking to the patients). Can you build logic into the forms that will prevent the wrong answers from being entered? Can changes to these forms result in tasks changing? Is it only certain forms that will be reviewed or will it be all of them? 

---

> **3. Will CHOs be physically with CHNs during a supervisory meeting or will the CHOs be doing this totally independently** 
CHOs duty is to supervise the information collected, so whenever new forms are submitted, they review them. CHOs meet CHNs physically from daily basis to weekly basis.

Do they review 100% of forms submitted? Or do they randomly choose forms to review? Is there a reason they don't access from the Reports tab instead of the Contacts tab? Where do they meet them... in the field or in the clinic?

---
> **6. And just to confirm, above you mentioned that this is primarily for convenience rather than an issue with accessing data in other areas, is that still correct?**
No, it's both. CHOs should not be able to see data in any other areas. For now, we only have two municpalities that's why it doesn't sound as inconvenient and data issue. But, we've scale up in June/July and then it'll be a major problem. The requirement is to see data under them only.

If it's a data privacy concern, then it wouldn't matter whether it is 2 or 7 municipalities. Is there something unique about the scenario today, with the 2 municipalities, that makes it OK for the CHO to see data in the other municipality?

**Comment by yrimal** [comment]
>If it's a data privacy concern, then it wouldn't matter whether it is 2 or 7 municipalities. Is there something unique about the scenario today, with the 2 municipalities, that makes it OK for the CHO to see data in the other municipality?

We discovered that this feature was lagging after going into production only and partner's first response was that one municipality person should not see other municipality's data. We convinced them by telling that we will work on it and will be available later. 

>Do they review 100% of forms submitted

The asked responsibility by this position is to review 100 % of the forms. 

>Is there a reason they don't access from the Reports tab instead of the Contacts tab?

Is only their report available if they access from Reports tab ? One of the reason they go from contacts tab could be easy to navigate ward by ward or CHNs by CHNs. 

>Where do they meet them... in the field or in the clinic?

In the field as well as Municipality office which is CHO's duty station. 

>Is it only certain forms that will be reviewed or will it be all of them?

All the forms will be reviewed. 

In general, we are applying and handling errors that can be handled on form themselves. Since CHOs meet CHNs and frequently go on field as well, they'll know information about patient or fchv area etc.

**Comment by michaelkohn** [comment]
> The asked responsibility by this position is to review 100 % of the forms.

How will the CHOs keep track of which forms they have already reviewed and which ones still need to be reviewed?

> Is only their report available if they access from Reports tab ? One of the reason they go from contacts tab could be easy to navigate ward by ward or CHNs by CHNs.

When you click on a report from a Contact, it actually takes you to the Reports tab anyway and that's the only place to go to edit them. So to go find the next report to review, will they stay on the Reports tab or navigate back to the Contact?




**_Can you share a screen recording of a CHO reviewing 2 or 3 reports?_**

**Comment by yrimal** [comment]
>How will the CHOs keep track of which forms they have already reviewed and which ones still need to be reviewed?

Hi @michaelkohn , I had a follow up with CHOs on how they keep track of the forms reviewed or not, you see that on reports tab there's a blue mark to denote whether a report is read (or opened) or not and it's easy for them to keep track of the forms they reviewed or not. 

CHO's main pain point is seeing reports that they do not need to see from other municipalities.

**Comment by nnbhandari1** [comment]
@michaelkohn, as we have deployed in one more municipality, 2 existing +1, and 2 more will be added this year. So by the end of July 2023 a total of 5 municipalities will have deployed the app.  We again received a request from the government to limit the access, i.e the users should not be allowed to access or modify places outside of their area. We need to prioritise this ASAP. Thanks
@yrimal @binokaryg @niraj21

**Comment by michaelkohn** [comment]
Hi @nnbhandari1 ... that's great, now that you have been up and running for some time, you should have some good data available. As mentioned before ([here](https://medic.slack.com/archives/C02GME3BQE6/p1659445996347739?thread_ts=1659260250.296459&cid=C02GME3BQE6), [here](https://medic.slack.com/archives/C4LVCKJJC/p1660767301078569?thread_ts=1660240749.725469&cid=C4LVCKJJC), [here](https://medic.slack.com/archives/C02GME3BQE6/p1660101393051489?thread_ts=1659260250.296459&cid=C02GME3BQE6), etc...), creating a new user type that is online/restricted would be a significant architectural change and unlikely to be developed without having high confidence in being able to achieve proportional and demonstrable impact, we don't have that yet. That said, I think there are many other low risk and quicker options, I'd be happy to prioritize working with the team to identify and explore those.

### When we were discussing this last year, we learned that...

1. There were no **specific** rules or guidance as to what kinds of errors the CHNs were supposed to look for when reviewing reports
2. CHNs are supposed to review **every** report submitted by CHOs, 500+ per day `*`
3. They manage this by utilizing the "read" flag on reports
4. There's no way to filter which ones they have reviewed and which ones they haven't reviewed
5. There's no way way to mark a report as "unread", so if you've clicked on a report accidentally, it gets marked as read
6. The "read" status is based on the user, not on the report, so the report might be read ("reviewed") for one person, but unread for another, making it difficult to triage those 500+ reports 

_`*` given an 8 hour day, this translates to reviewing 60+ reports per hour = 1 report every minute_

### Now that they’ve been up and running for a while...
1. Has the workflow evolved at all? Are they doing the same thing?
2. Do you have any more clarity on the (lack of) rules mentioned in point 1?
3. Are the volumes mentioned in point 2 still accurate? 
4. Have you analyzed the live data to better understand what types of things actually are being updated most often, how many reports are being "read", actual volume of reports per CHN, etc...?

### Have you explored these options further...

Last year we:
1. Suggested [an option](https://medic.slack.com/archives/C02GME3BQE6/p1659372319568429?thread_ts=1659260250.296459&cid=C02GME3BQE6) that would require some config changes but no cht-core development 
2. [Proposed and prototyped](https://medic.slack.com/archives/C4LVCKJJC/p1660240749725469) a cht-core change that would be pretty quick to develop and restrict what data users see by default 
3. Developed the [Data Observation Toolkit](https://github.com/medic/Data-Observation-Toolkit), which could be used to identify bad data rather than having the CHN do it.

Have any of these been tried or considered in more depth?

**Comment by kitsao** [comment]
Adding a MoH Kenya eCHIS scenario

#### User story
As a CHA supervising at least one CU, I want to be able to access all CUs using one login account. Currently, I am forced to logout and login with other credentials specific to the CU I wish to supervise so that I can access in app analytics and tasks.

#### Background

- Currently, CHAs and CHVs are offline users. The ideal relationship is that one CHA supervises one CU. However, there are several instances of one CHA supervising more than one CU.
- In a 1 to many CHA to CU mapping, CHAs have multiple credentials, according to the number of CUs. However, logging out and in when switching CUs to supervise is not user friendly.

#### Exploration

- Group the many CUs under one CHA into a region, essentially introducing another hierarchy level, CU -> CHA Region.
- CHA to belong to CHA region and not CU.

#### Challenge

- In cases where the number of CUs grows to say 5+, the number of documents accumulates rapidly.
- Number of docs at CHA region = 5+ CUs * 10 CHV areas * 100+ households * 5+ household members = 25000
- The number of docs will easily hit the limits as reports stream in hence rendering the app non performant.

#### What if

- Support for restricting online users to their places would be a first step towards mitigating the docs exponential growth, with support for [online tasks and targets](https://forum.communityhealthtoolkit.org/t/tasks-for-online-users/574/4) being the next.

cc @freddieptf @IreneAchola

**Comment by dianabarsan** [comment]
It looks like CouchDb team is working on per-document access rules for online users: https://github.com/apache/couchdb/pull/5842

This could be an elegant and simple solution to this issue.
